components/TopBarProfileMenu.js

```diff
@@ -75,29 +135,40 @@ class TopBarProfileMenu extends React.Component {
     if (key === 'Escape' || key === 'Esc' || keyCode === 27) {
       this.setState({
         showProfileMenu: false,
+        showMyAccount: false,
       });
     }
   };
 
   logout = () => {
-    this.setState({ showProfileMenu: false, status: 'loggingout' });
+    this.setState({ showProfileMenu: false, showMyAccount: false, status: 'loggingout' });
     this.props.logout();
     this.setState({ status: 'loggedout' });
   };
 
   onClickOutside = () => {
-    this.setState({ showProfileMenu: false });
+    this.setState({ showProfileMenu: false, showMyAccount: false });
   };
 
   toggleProfileMenu = e => {
-    this.setState(state => ({ showProfileMenu: !state.showProfileMenu }));
+    this.setState(state => ({ showProfileMenu: !state.showProfileMenu, showMyAccount: false }));
     // don't propagate to onClickOutside
     e.nativeEvent.stopImmediatePropagation();
   };
 
+  showAccountInfo = e => {
+    this.setState({ showMyAccount: true });
+    e.stopPropagation();
+  };
+
+  hideAccountInfo = e => {
+    this.setState({ showMyAccount: false });
+    e.stopPropagation();
+  };
+
```

Comment:
Suggest having one function; `showAccountInfo` with a boolean argument to show or hide the account info. 🐨

Reply:
It does not need a toggle I guess, one option for opening it and another for closing it

Reply:
Maybe I can go for inline function on click event?

Reply:
@naman360: What I was thinking is along the lines of, 

```
showAccountInfo = (e, showMyAccount) => {
    this.setState({ showMyAccount });
    e.stopPropagation();
};
```
And then you can do, `onClick={(e) => this.showAccountInfo(e, true)}` for places where you used `onClick={this.showAccountInfo}` and `onClick={(e) => this.showAccountInfo(e, false)}` for places where you used `onClick={this.hideAccountInfo}`. Hope that makes sense. Let me know if that works and if you have any further questions. It's a minor thing to prevent using two functions; but I don't have a strong opinion on this so if you prefer it's okay to keep both functions as well. 😄